src/CaptivePortalInstance.cpp

```diff
@@ -158,6 +158,13 @@ void CaptivePortalInstance::handleWebSocketClientConnected(std::uint8_t socketId
   }
 
   Serialization::Local::SerializeReadyMessage(connectedNetworkPtr, GatewayConnectionManager::IsLinked(), std::bind(&CaptivePortalInstance::sendMessageBIN, this, socketId, std::placeholders::_1, std::placeholders::_2));
+
+  // Send all previously scanned wifi networks
+  auto& networks = OpenShock::WiFiManager::GetDiscoveredWiFiNetworks();
+
+  for (auto& network : networks) {
+    Serialization::Local::SerializeWiFiNetworkEvent(Serialization::Types::WifiNetworkEventType::Discovered, network, std::bind(&CaptivePortalInstance::sendMessageBIN, this, socketId, std::placeholders::_1, std::placeholders::_2));
+  }
```

Comment:
maybe you can try to update the schema for DeviceToLocal.fbs to include an array instead?
This might improve the stability and efficiency of the esp and frontend